client/main.lua

```diff
@@ -294,6 +294,11 @@ local function startTestDrive(vehModel)
     TriggerServerEvent('qbx_vehicleshop:server:customTestDrive', vehModel, playerId)
 end
 
+lib.onCache('vehicle', function(value)
+    if value or not LocalPlayer.state.inTestDrive then return end
+    LocalPlayer.state:set('inTestDrive', nil, true)
+end)
```

Comment:
Is there some point in time at the beginning of a test drive where the inTestDrive state has been set but you aren't in the vehicle yet?